### PR TITLE
Improving Typescript developer experience

### PR DESCRIPTION
**Pull request by code-ape**
## Overview
This is a small PR to improve the developer experience of this repo due to recent introduction of Typescript. This is from conversation with @OlivierCavadenti and @kibertoad.

## Related Issues & PRs
1. #5144

## Changes
1. Add `npm run clean` command to delete all ignored files in `lib/` which should be all compiler outputs of Typescript files.
2. Add npm `postinstall` script to automatically build Typescript files each time a developer installs or sets up the repo. 
3. Improved developer documentation in `CONTRIBUTING.md`.

## Open Questions
- [x] Should there be a new developer documentation section added in `README.md`?
    - Answer from @kibertoad: Not for the time being.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -171,11 +171,11 @@ psql -h localhost -U postgres -d knex_test
...
 
-To develop in this repository use the `npm run build` command. If you wish to have the `tsc` compiled watch and recompile on changes then run `npm run build:ts -- --watch`. Note that for easy integration with Javascript the outputted files are done in a "side-by-side" manner meaning that `lib/foo/bar.ts` will result in `lib/foo/bar.js`.
+To develop in this repository use the `npm run build` and `npm run clean` commands to compile and delete the `.js` and related files from `.ts` files. If you wish to have the `tsc` compiled watch and recompile on changes then run `npm run build:ts -- --watch`. Note that for easy integration with Javascript the outputted files are done in a "side-by-side" manner meaning that `lib/foo/bar.ts` will result in `lib/foo/bar.js`. This is automatically via the npm `postinstall` script trigger which files when you run commands like `npm install` or `npm ci`, thus making it easier to not have to worry about this if you're working in pure Javascript.
```
Comment (kibertoad) [inline CONTRIBUTING.md:178]: `This is automatically via the npm `postinstall` script trigger which files when you run commands `

some words seem to be missing

Comment (code-ape) [reply CONTRIBUTING.md]: Good catch @kibertoad! Was typing a bit faster than I was thinking 😅 just pushed corrections. Also, see my question at the bottom of the PR description related to `README.md`.

Comment (kibertoad) [reply CONTRIBUTING.md]: let's keep it like this for now and see

**Review by kibertoad — APPROVED** [review]

**Comment by code-ape** [comment]
Ready for merge at your convenience, @kibertoad 😄

**Review by kibertoad — APPROVED** [review]

**Pull request merged**